spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/HqlQueryTransformer.java

```diff
@@ -164,6 +170,33 @@ public List<JpaQueryParsingToken> visitOrderedQuery(HqlParser.OrderedQueryContex
 		return tokens;
 	}
 
+	/**
+	 * Determine when an {@link org.springframework.data.domain.Sort.Order} parameter should alias (or not).
+	 *
+	 * @param order
+	 * @return boolean whether or not to apply the primary FROM clause's alias
+	 */
+	private boolean shouldAlias(Sort.Order order) {
+
+		if (orderParameterIsAFunction(order)) {
+			return false;
+		}
+
+		if (orderParameterReferencesAProjectionAlias(order)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private boolean orderParameterIsAFunction(Sort.Order order) {
+		return order.getProperty().contains("(");
```

Comment:
How is this even remotely possible? While properties are string, do we rewrite the sort to contain a function?